src/services/auth.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, HttpException, HttpStatus, Body } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import * as bcrypt from 'bcrypt';
+import * as jwt from 'jsonwebtoken';
+
+import { Model } from 'mongoose';
+import { User } from '../interfaces/user.interface';
+import { SingUpDto } from 'src/validation/signUp.dto';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectModel('User') private readonly userModel: Model<User>,
+    private readonly configService: ConfigService,
+  ) {}
+  async signup(
+    @Body()
+    signUpDto: SingUpDto,
+  ): Promise<any> {
```

Comment:
Using any type here nullifies the use of TypeScript, we can define an interface for it as well.